### PR TITLE
[Build System: build-script] Remove the tar module from swift_build_support.

### PR DESCRIPTION
**Pull request by Rostepher**
The single function it provided to archive a source diretory has been moved to build-script which was the only caller.

### CODE REVIEW

**Comment by Rostepher** [comment]
@swift-ci please smoke test

**Review by compnerd — APPROVED** [review]
I'm fine with the shuffling, not the original code.  I think that we should get this merged as-is and then follow it up with a replacement with `tarfile`.

**Comment by Rostepher** [comment]
@compnerd I just wanted to remove this unnecessary module while I'm transitioning the important functionality from `swift_build_support` to `build_swift`. I'm definitely in favor of using a Python standard than rolling our own wrapper around the `tar` CLI.

File: utils/build-script [hunk 1]
```diff
@@ -89,6 +88,25 @@ def print_xcodebuild_versions(file=sys.stdout):
     file.flush()
 
 
+def tar(source, destination):
+    """
+    Create a gzip archive of the file at 'source' at the given
+    'destination' path.
+    """
+    # We do not use `tarfile` here because:
+    #  - We wish to support LZMA2 compression while also supporting Python 2.7.
```
Comment (Rostepher) [inline utils/build-script]: Pretty sure we've never used this for LZMA2 compression, thus we should just use the builtin `tarfile` library. I'll do that in a follow-up PR.

Comment (compnerd) [reply utils/build-script]: The builtin `tarfile` module supports lzma as well I *think*

Comment (Rostepher) [reply utils/build-script]: Only for Python 3.2 (or 3.3) and beyond. Python 2.7 supports gzip however, which appears to be all we use for the time being.

**Pull request merged**